lib/domain.js

```diff
@@ -68,12 +68,27 @@ exports.active = null;
 
 inherits(Domain, EventEmitter);
 
+function setHasErrorListener(type) {
+  if (type === 'error')
+    this._hasErrorListener = true;
+}
+
+function removeHasErrorListener(type) {
+  if (type === 'error')
+    this._hasErrorListener = false;
```

Comment:
I think this should either be a counter, or test whether `.listenerCount('error') === 0`? It seems possible that there could be two error listeners on a domain

Reply:
Updated. My hope is we can introduce `hasListener` helper for these use cases (which is included in an earlier commit). We've previously accessed _events directly for this purpose and `listenerCount` does a bit more work than necessary.